# Temperature Conversion

As part of this colab, I'm going to show you how to create python functions to convert Celsius to Farenheit, and then back again.

Then I'm going to give it back to you and have you figure out how to Celsius to Kelvin, and same with Farenheit to Kelvin.

## Why Colab and Why Python?

Colab, short for Google Colaboratory, is a free cloud-based service that lets you write and run Python code in your browser.  It is VERY VERY popular for Data Scientists, ML Coders, and also Classroom settings.

## Background

Here are the formulas you need to know to convert between the two temperature scales.

### Farenheit to Celsius

$f = (c*9/5) + 32$


### Celsius to Farenheit

$c = (f-32)*(5/9) $




## Example Starter Code

### Convert Farenheit to Celsius

In [ ]:
F = 82.4 # @param {"type":"number"}

def f_to_c(f):
  return (f-32)/1.8

f_to_c(F)

28.000000000000004

#### Test the function




In [ ]:
assert f_to_c(32.0) == 0
assert f_to_c(212.0) == 100
assert f_to_c(82.4) == 28.000000000000004

### Convert Celsius to Farenheit

In [ ]:
C = 100 # @param {"type":"number"}

def c_to_f(c):
  return (c*1.8) + 32

c_to_f(C)

212.0

#### Test the function

In [ ]:
assert c_to_f(100) == 212.0
assert c_to_f(0) == 32
assert c_to_f(28) == 82.4

## Your Turn

Now it's your time to figure out how to do this yourself.



### Kelvin to Celsius

$ C = K - 273.15$

### Kelvin to Farenheit

$F = (K-273.15) * (9/5) + 32$

### Testing Data

| Farenheit | Celsius | Kelvin |
|-----------|---------|--------|
| 32        | 0       | 273.15 |
| 212       | 100     | 373.15 |
| 82.4      | 28      | 301.15 |


### Convert Celsius to Kelvin

In [ ]:
K = 100 # @param {"type":"number"}


212.0

#### Test the function

### Convert Farenheit to Kelvin

In [ ]:
K = 100 # @param {"type":"number"}


212.0

#### Test the function